tests/unit/oom-score-adj.tcl

```diff
@@ -73,5 +82,21 @@ if {$system_name eq {linux}} {
                 assert {[r config get oom-score-adj-values] == {oom-score-adj-values {0 100 100}}}
             }
         }
+
+        test {CONFIG SET oom-score-adj-values doesn't touch proc when disabled} {
+            set orig_osa [get_oom_score_adj]
+            
+            set other_val1 [expr $orig_osa + 1]
+            set other_val2 [expr $orig_osa + 2]
+            
+            r config set oom-score-adj no
+            
+            set_oom_score_adj $other_val2
+            assert_equal [get_oom_score_adj] $other_val2
+
+            r config set oom-score-adj-values "$other_val1 $other_val1 $other_val1"
+            
+            assert_equal [get_oom_score_adj] $other_val2
+        }
```

Comment:
in the other PR i listed 3 tests that could be added
1. to make sure that when disabling oom-score-adj, it reverts what it did when it was enabled.
2. make sure that setting either of these two configs when oom-score-adj is disabled, it will not try to modify /proc and fail on non-Linux systems.
3. consider improving the "special" oom-score-adj-values config to detect no-op changes and skip them.

IIUC you took care of 2.
i think we should at least do 1 too.
arguably 3 is not a must (but note that we **can* make a test for it by having tcl modify proc behind the back of redis).

Reply:
1. Test 1 is covered by the _"CONFIG SET rollback on apply error"_ test in _introspection_. This is the test that failed originally because rollback didn't recover correctly the oom. If you want I can add another explicit test, but it's indeed covered there.
2. The purpose of the test is not only to validate we don't fail on non-linux systems but also to validate that on linux systems we don't touch _/proc_ when `oom-score-adj` is disabled. But yes, it's handled by me test.
3. This isn't a test it's a general improvement, right? I'll look into it.

Reply:
regarding 1, i think we want an explicit test for that, and not rely on the config infrastructure test. it's a simple test, let's have it.